builtin/providers/aws/resource_aws_ses_receipt_rule.go

```diff
@@ -0,0 +1,730 @@
+	"github.com/hashicorp/terraform/helper/schema"
+	"sort"
+)
+
+func resourceAwsSesReceiptRule() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsSesReceiptRuleCreate,
+		Update: resourceAwsSesReceiptRuleUpdate,
+		Read:   resourceAwsSesReceiptRuleRead,
+		Delete: resourceAwsSesReceiptRuleDelete,
+
+		Schema: map[string]*schema.Schema{
+			"name": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"rule_set_name": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"after": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+
+			"enabled": &schema.Schema{
+				Type:     schema.TypeBool,
+				Optional: true,
+				Computed: true,
+			},
+
+			"recipients": &schema.Schema{
+				Type:     schema.TypeSet,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+				Optional: true,
+				Set:      schema.HashString,
+			},
+
+			"scan_enabled": &schema.Schema{
+				Type:     schema.TypeBool,
+				Optional: true,
+				Computed: true,
+			},
+
+			"tls_policy": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+
+			"add_header_action": &schema.Schema{
+				Type:     schema.TypeSet,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"header_name": &schema.Schema{
+							Type:     schema.TypeString,
+							Required: true,
+						},
+
+						"header_value": &schema.Schema{
+							Type:     schema.TypeString,
+							Required: true,
+						},
+
+						"position": &schema.Schema{
+							Type:     schema.TypeInt,
+							Required: true,
+						},
+					},
+				},
+				Set: func(v interface{}) int {
+					var buf bytes.Buffer
+					m := v.(map[string]interface{})
+					buf.WriteString(fmt.Sprintf("%s-", m["header_name"].(string)))
+					buf.WriteString(fmt.Sprintf("%s-", m["header_value"].(string)))
+					buf.WriteString(fmt.Sprintf("%d-", m["position"].(int)))
+
+					return hashcode.String(buf.String())
+				},
+			},
+
+			"bounce_action": &schema.Schema{
+				Type:     schema.TypeSet,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"message": &schema.Schema{
+							Type:     schema.TypeString,
+							Required: true,
+						},
+
+						"sender": &schema.Schema{
+							Type:     schema.TypeString,
+							Required: true,
+						},
+
+						"smtp_reply_code": &schema.Schema{
+							Type:     schema.TypeString,
+							Required: true,
+						},
+
+						"status_code": &schema.Schema{
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+
+						"topic_arn": &schema.Schema{
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+
+						"position": &schema.Schema{
+							Type:     schema.TypeInt,
+							Required: true,
+						},
+					},
+				},
+				Set: func(v interface{}) int {
```

Comment:
in the Set func, what happens to the hash if some of the optional values are not passed?

I think the set func should be something like this:

```
if d, ok := m["topic_arn"]; ok {
		buf.WriteString(fmt.Sprintf("%s-", m["topic_arn"].(string)))
}
```